src/reader/mod.rs

```diff
@@ -1930,3 +1934,74 @@ mod test {
     pub(super) use check;
     pub(super) use small_buffers;
 }
+
+fn read_to_string_impl<'b, R: BufRead>(
+    starting_buffer_position: usize,
+    reader: &mut R,
+    parser: Parser,
+    end: QName,
+    temporary_buf: &mut Vec<u8>,
+    output_buf: &'b mut Vec<u8>,
+) -> (Result<&'b str>, Parser) {
+    // Initialize a new XMLParser
+    let reader = StoredReader::new(reader, starting_buffer_position, output_buf);
+    let mut temp_reader: Reader<StoredReader<&mut R>> = Reader { reader, parser };
+
+    let range = temp_reader.read_to_end_into(end, temporary_buf);
+
+    let Reader { reader, parser } = temp_reader;
+    let as_str = range.and_then(|r| reader.get_stored_value(r.start, r.end));
+    (as_str, parser)
+}
+
+/// Wraps an existing Reader and stores all the bytes read out of the `inner`.
+struct StoredReader<'a, R> {
+    inner: R,
+    initial_offset: usize,
+    stored: &'a mut Vec<u8>,
+}
+
+impl<'a, R: Read> Read for StoredReader<'a, R> {
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+        let result = self.inner.read(buf);
+        if let Ok(size) = &result {
+            self.stored.extend_from_slice(&buf[0..*size]);
+        }
+        result
+    }
+}
+
+impl<'a, R> StoredReader<'a, R> {
```

Comment:
I think, that both methods can be inlined in their place of use. They a used only once and they both trivial